### PR TITLE
AzureStack: expose configurable vm image alias doc

### LINKED ISSUE #3338: Azurestack: az vm images list  returns a cached response which is not applicable for azurestack
### Description
az vm images list right now lists a cached response. The images listed in this are not likely available in azure stack instance.

In azure stack the admin uploads the images as part of the initial setup and he may upload only a specific set of images.

az vm images list --all actually makes the api call and gets the right set of images. May be the solution is to make --all as the default and use the cached data only for TabTab completion

### LINKED ISSUE #3340: AzureStack: images aliases is not applicable for azurestack
### Description
CLI has a set of aliases for the images referred from the following location
https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-compute/quickstart-templates/aliases.json

These aliases makes the assumption that these images are available in all the clouds and matches the 
definition like below
          "UbuntuLTS":{
            "publisher":"Canonical",
            "offer":"UbuntuServer",
            "sku":"16.04-LTS",
            "version":"latest"
          }

In azure stack , the alias does not make much sense as the admin may not have uploaded the same image
in azure stack the Ubuntu server sku could be 14.04.3 instead of 16.04-LTS

luckily , the image parameter can be specified with the whole urn like below and this is what being used in azurestack 
Canonical:UbuntuServer:14.04.3-LTS:1.0.0

All the public documentation is referring to the use of aliases and the aliases most likely wont work against azurestack unless the admin chooses to upload image matching the azure definitions

### PR DESCRIPTION
**Pull request by yugangw-msft**
@derekbekoe, please review and let me know whether I did the things in a right way
This fix #3340, also fix #3338
### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/cloud.py [hunk 1]
```diff
@@ -204,6 +204,10 @@ def __str__(self):
 
 
 KNOWN_CLOUDS = [AZURE_PUBLIC_CLOUD, AZURE_CHINA_CLOUD, AZURE_US_GOV_CLOUD, AZURE_GERMAN_CLOUD]
+for kc in KNOWN_CLOUDS:
+    # pylint: disable=line-too-long
+    setattr(kc.endpoints, 'vm_image_alias_doc',
+            'https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-compute/quickstart-templates/aliases.json')
```
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/cloud.py]: Can you set it individually for each known cloud?
I prefer all the endpoints to be set on the constructors for each `Cloud` class.
https://github.com/Azure/azure-cli/blob/master/src/azure-cli-core/azure/cli/core/cloud.py#L149

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py [hunk 2]
```diff
@@ -72,8 +72,13 @@ def _load_images_from_publisher(publisher):
 
 
 def load_images_from_aliases_doc(publisher=None, offer=None, sku=None):
-    target_url = ('https://raw.githubusercontent.com/Azure/azure-rest-api-specs/'
-                  'master/arm-compute/quickstart-templates/aliases.json')
+    from azure.cli.core.cloud import get_active_cloud
+    CLOUD = get_active_cloud()
+    target_url = getattr(CLOUD.endpoints, 'vm_image_alias_doc', None)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py]: You should be able to do `CLOUD.endpoints. vm_image_alias_doc ` then catch the `CloudEndpointNotSetException` due to https://github.com/Azure/azure-cli/blob/master/src/azure-cli-core/azure/cli/core/cloud.py#L82.

Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py]: done

File: src/azure-cli-core/azure/cli/core/cloud.py (hunk 1 above)
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/cloud.py]: sure, done

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py [hunk 3]
```diff
@@ -72,8 +72,13 @@ def _load_images_from_publisher(publisher):
 
 
 def load_images_from_aliases_doc(publisher=None, offer=None, sku=None):
-    target_url = ('https://raw.githubusercontent.com/Azure/azure-rest-api-specs/'
-                  'master/arm-compute/quickstart-templates/aliases.json')
+    from azure.cli.core.cloud import get_active_cloud, CloudEndpointNotSetException
+    CLOUD = get_active_cloud()
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py]: Would suggest not capitalizing it.

**Pull request merged**